packages/react/src/hooks/notifyStateManagement.ts

```diff
@@ -0,0 +1,106 @@
+import type { NotifyClientTypes } from "@walletconnect/notify-client";
+import { useSubscriptionState } from "@web3inbox/core";
+import { useCallback, useEffect, useState } from "react";
+import { useWeb3InboxClient } from "./web3inboxClient";
+
+export const useMessages = (account?: string) => {
+  const client = useWeb3InboxClient();
+  const { messages: messagesTrigger } = useSubscriptionState();
+  const [messages, setMessages] = useState<
+    NotifyClientTypes.NotifyMessageRecord[]
+  >(client?.getMessageHistory(account) ?? []);
+
+  useEffect(() => {
+    if (!client) return;
+
+    setMessages(client.getMessageHistory(account));
+  }, [client, messagesTrigger, account]);
+
+  const deleteMessage = useCallback(
+    async (id: number) => {
+      if (client) {
```

Comment:
Nit: Not sure what would happen if an `id` wasn't provided. Should we safe guard it ?
```suggestion
      if (client && id) {
```